rocksdb_replicator/replicated_db.cpp

```diff
@@ -152,7 +195,7 @@ void RocksDBReplicator::ReplicatedDB::pullFromUpstream() {
   std::weak_ptr<ReplicatedDB> weak_db = shared_from_this();
   auto options = rpc_options_;
   client_->future_replicate(options, req).via(executor_)
-    .then([weak_db = std::move(weak_db)] (folly::Try<ReplicateResponse>&& t) {
+    .then([weak_db = std::move(weak_db), this] (folly::Try<ReplicateResponse>&& t) {
```

Comment:
Since `resetUpstream()` is part of the class, I needed to capture `this`. Do you have a suggestion on a better way to do it? Thanks for the info about the weak_ptr

Reply:
Oh, I guess I can do `db->resetUpstream()` instead.